homeassistant/components/ondilo_ico/sensor.py

```diff
@@ -0,0 +1,185 @@
+"""Platform for sensor integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from ondilo import OndiloError
+
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+SENSOR_TYPES = {
+    "temperature": [
+        "Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
```

Comment:
We don't set a custom icon when we set a device class. Those will already have specific icons in the frontend.

Reply:
Done in PR https://github.com/home-assistant/core/pull/44837